### ISSUE TITLE
Ability to rename window.__NUXT__ context object

### ISSUE THREAD

**Issue by mvanalphen** [nuxt/nuxt#18870]
### Describe the feature

In Nuxt 2, it was possible to rename the `__NUXT__` context object assigned to the `window` through the `globals.context` option in `nuxt.config.js`, as documented [here](https://nuxtjs.org/docs/configuration-glossary/configuration-global-name/#the-globals-property).

In Nuxt 3, it seems that it is not possible to rename this context object through the Nuxt configuration. However, in [this pull request](https://github.com/nuxt/framework/pull/8883), the possibility of changing the `rootId` and `rootTag` was re-added, which was previously possible through the aforementioned `globals` configuration option. 

As such, the feature request is to allow clients to rename the `__NUXT__` context object through their Nuxt configuration.

### Context
For context: I'm currently experimenting with loading a new Nuxt application within an existing Nuxt application. The reason for this is that we run a number of micro-frontends as separate Nuxt projects. When a user clicks on an element that exists in some of these Nuxt micro-frontends, we want to show a slide-in component which is actually a separately maintained Nuxt or Vue project. We're still running some experiments in how we want to handle this, with one option being loading an entirely new Nuxt application.

An issue with loading a new Nuxt application, is that the micro-frontends naturally define their own `__NUXT__` context, while the dynamically loaded project will also want to do the same. These  will conflict with each other. As such, the dynamically loaded project would want to use its own `window.__NUXT__` context object, e.g. `__NUXT_EXAMPLE__`.

This is all still in very early stages and it may be a pretty bad idea, and I may end up just building this project in vanilla Vue 3. However, I figured that it may be worthwhile to re-add this functionality from Nuxt 2. 

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by mvanalphen** [comment]
Should anybody come across this from Google:

As a workaround during for my experimentation, I'm temporarily resolving this by extending the Vite and Nitro configs in `nuxt.config.ts` by adding the `'@rollup/plugin-replace` plugin, like so:

``` typescript
import type { OutputOptions, Plugin } from 'rollup';
import replace from '@rollup/plugin-replace';

const replacePlugin = replace({
  __NUXT__: '__NUXT_EXAMPLE__',
});

export default defineNuxtConfig({
  hooks: {
    'vite:extendConfig': (viteInlineConfig, { isClient }) => {
      if (!isClient) {
        return;
      }

      const rollupOptions = viteInlineConfig.build?.rollupOptions;
      if (!rollupOptions) {
        throw Error("Could not find Vite's Rollup options");
      }

      if (rollupOptions.plugins) {
        const plugins = rollupOptions.plugins as Plugin[];
        plugins.push(replacePlugin);
      } else {
        rollupOptions.plugins = [replacePlugin];
      }

      const output = rollupOptions?.output as OutputOptions;
      if (!output) {
        throw Error("Could not find the output of Vite's Rollup options");
      }
    },

    'nitro:config': nitroConfig => {
      const plugins = nitroConfig.rollupConfig?.plugins as Plugin[];
      if (!plugins) {
        throw Error("Could not find the plugins of Nitro's Rollup config");
      }

      plugins.push(replacePlugin);
    },
  },
});
```

**Comment by mvanalphen** [comment]
The `experimental.noScripts` option in `nuxt.config.ts` will remove the context object entirely, and also solves this problem.

**Comment by Baroshem** [comment]
@danielroe 

I looked at the description and it seems to me that there is no need for additional work here. If I am mistaken, please let me know. I could take a look at this issue :)

**Comment by mvanalphen** [comment]
It's worth noting that `experimental.noScripts` does limit the Nuxt application in the sense that there'll be no scripts generated using `nuxi generate`, so that'll definitely not suit a lot of use cases. There is a workaround available, but it is not ideal.

**Comment by danielroe** [comment]
We're not inclined to support renaming `__NUXT__` on its own but we absolutely want to support multi-app, which we can track in https://github.com/nuxt/nuxt/issues/21635.

**Comment by victorgarciaesgi** [comment]
HI @mvanalphen , even if this issue is closed, did you encountered more issues? 
Is what you linked for the `window.__NUXT__` was the only blocker into enabling a CSR nuxt app into a micro-frontend?
We have the same case and would love to know! :)

**Comment by victorgarciaesgi** [comment]
Sorry to bother you again @mvanalphen , but did you by chance encountered a `nuxt context error` ?
![image](https://github.com/nuxt/nuxt/assets/15092120/86ae8a56-5d10-4808-ada5-5c7a045d50b1)

The rollup replace plugin worked, and I have the 2nd app loaded in the first one dynamically (user action)

But it seems when mounting Nuxt still have 2 instances cohabiting, is it something your nitro plugin prevented? 

BTW: I'm in dev mode with a different build for MF, enabled the rollup plugin in Vite dev too. I saw the `experimental.noScripts` is only working on a full build and not in dev so maybe it's related
Thanks :D

**Comment by victorgarciaesgi** [comment]
I managed to made it working in dev without the nitro plugin.
It's super hacky for now as i'm patching some lines of Nuxt core to avoid the Nuxt context conflict.

```ts
const replacePlugin: any = replace({
    delimiters: ['', ''],
    values: {
        __NUXT__: '__NUXT_MF__',
        [`getContext("nuxt-app")`]: `getContext("nuxt-app-mf")`,
    },
});
```

Had to remove the `baseUrl` of the app and update the `buildAssetsDir` property too.
Deactivated the `pages` property also.
To handle the redirection I used `routeRules` like so 

```ts
routeRules: {
        '/mf/project-client': {redirect: '/'},
}
```

Still a lot of hacky stuff, working on dev, not tested on prod build yet

Thanks again for your help and infos @mvanalphen :D
